src/Illuminate/Foundation/Events/DiscoverEvents.php

```diff
@@ -88,11 +88,13 @@ protected static function getListenerEvents($listeners, $basePath)
     protected static function classFromFile(SplFileInfo $file, $basePath)
     {
         $class = trim(Str::replaceFirst($basePath, '', $file->getRealPath()), DIRECTORY_SEPARATOR);
+        $composer = json_decode(file_get_contents(app()->basePath().DIRECTORY_SEPARATOR.'composer.json'), true);
```

Comment:
can't jank to call the app helper here

Reply:
What should I do instead? The `app()` helper is called line 95 and 96 below too